src/dev/impl/DevToys/ViewModels/Tools/Text/XmlValidator/XmlValidatorToolViewModel.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Composition;
+using System.Xml.Schema;
+using DevToys.Api.Core.Settings;
+using DevToys.Api.Tools;
+using DevToys.Core.Threading;
+using DevToys.Models;
+using DevToys.ViewModels.Tools.XmlValidator.Parsing;
+using DevToys.Views.Tools.XmlValidator;
+using Microsoft.Toolkit.Mvvm.ComponentModel;
+using Microsoft.UI.Xaml.Controls;
+
+namespace DevToys.ViewModels.Tools.XmlValidator
+{
+    [Export(typeof(XmlValidatorToolViewModel))]
+    public class XmlValidatorToolViewModel : ObservableRecipient, IToolViewModel
+    {
+        private string? _xsdSchemaData;
+        private string? _xmlData;
+        private XmlParsingResult? _parsedXml;
+        private XsdParsingResult? _parsedXsdScheme;
+        private readonly XmlValidatorStrings _localizedStrings;
+        private InfoBarData? _validationResult;
+
+        internal ISettingsProvider SettingsProvider { get; }
+        internal XmlValidatorStrings Strings => LanguageManager.Instance.XmlValidator;
+
+        internal string? XsdSchema
+        {
+            get => _xsdSchemaData;
+            set
+            {
+                ThreadHelper.ThrowIfNotOnUIThread();
+                SetProperty(ref _xsdSchemaData, value);
+                ProcessNewXsdData();
+            }
+        }
+        
+        internal string? XmlData
+        {
+            get => _xmlData;
+            set
+            {
+                ThreadHelper.ThrowIfNotOnUIThread();
+                SetProperty(ref _xmlData, value, broadcast: true);
+                ProcessNewXmlData();
+            }
+        }
+
+        public InfoBarData? ValidationResult
+        {
+            get => _validationResult;
+            private set => SetProperty(ref _validationResult, value);
+        }
+        
+        public Type View { get; } = typeof(XmlValidatorToolPage);
+        
+        [ImportingConstructor]
+        public XmlValidatorToolViewModel(ISettingsProvider settingsProvider)
+        {
+            SettingsProvider = settingsProvider;
+            _localizedStrings = new XmlValidatorStrings();
+        }
```

Comment:
I noticed that when the tool is opened the first time, the info bar is hidden until we type something in one of the code editor.

![image](https://user-images.githubusercontent.com/3747805/173728069-d81542e8-288c-4c47-ba1b-30df7a2a70a3.png)

The UI feels a bit empty at the bottom. I'd suggest to call `DisplayValidationResult()` in the constructor to show the info bar immediately. It provides a good information to the user in my opinion.

Reply:
Done!
The `InfoBar` should always be displayed now.